lib/include/openamp/virtio.h

```diff
@@ -121,7 +121,6 @@ struct virtio_device {
 	uint32_t index; /**< unique position on the virtio bus */
 	struct virtio_device_id id; /**< the device type identification
 				         (used to match it with a driver). */
-	struct metal_spinlock lock; /**< spin lock */
```

Comment:
if we remove this lock how can we guarantee there is race condition in accessing the vrings? I think we should make use of the lock in the virtio access

Reply:
lock is not used in implementation. A lock is already handled by rpmsg client. Do you think we need both?

Reply:
when we use virtio, shall we always assume that who uses the virtio will make sure no race condition, maybe this is the right assumption